sdk/sdk-view/src/test/java/mil/emp3/api/SquareTest.java

```diff
@@ -0,0 +1,287 @@
+                       null,
+                       new GeoStrokeStyle(),
+                       null,
+                       new GeoLabelStyle(),
+                       false,
+                       true,
+                       0.0,
+                       0.0,
+                       null,
+                       false,
+                       Collections.EMPTY_LIST,
+                       false,
+                       Collections.EMPTY_LIST,
+                       "",
+                       "",
+                       "",
+                       new HashMap());
+    }
+
+    @Test
+    public void geoPosConstructor() {
+        final GeoPosition gp = new GeoPosition();
+        final GeoSquare gs = new GeoSquare();
+        gs.setPositions(Collections.singletonList(gp));
+        gs.setFillStyle(null);
+        s1 = new Square(gp);
+        gs.setTimeStamp(s1.getTimeStamp());
+        validateSquare(s1,
+                gs,
+                100.0,
+                FeatureTypeEnum.GEO_SQUARE,
+                Collections.EMPTY_LIST,
+                Collections.EMPTY_LIST,
+                Collections.EMPTY_LIST,
+                Collections.singletonList(gp),
+                gs.getTimeStamp(),
+                Collections.EMPTY_LIST,
+                null,
+                new GeoStrokeStyle(),
+                null,
+                new GeoLabelStyle(),
+                false,
+                true,
+                0.0,
+                0.0,
+                gp,
+                false,
+                Collections.EMPTY_LIST,
+                false,
+                Collections.EMPTY_LIST,
+                "",
+                "",
+                "",
+                new HashMap());
+    }
+
+    @Test
+    public void geoPosWidthConstructor() {
+        final GeoPosition gp = new GeoPosition();
+        s1 = new Square(gp, 5.4);
+        final GeoSquare gs = new GeoSquare();
+        gs.setPositions(Collections.singletonList(gp));
+        gs.setWidth(5.4);
+        gs.setFillStyle(null);
+        gs.setTimeStamp(s1.getTimeStamp());
+        validateSquare(s1,
+                gs,
+                5.4,
+                FeatureTypeEnum.GEO_SQUARE,
+                Collections.EMPTY_LIST,
+                Collections.EMPTY_LIST,
+                Collections.EMPTY_LIST,
+                Collections.singletonList(gp),
+                gs.getTimeStamp(),
+                Collections.EMPTY_LIST,
+                null,
+                new GeoStrokeStyle(),
+                null,
+                new GeoLabelStyle(),
+                false,
+                true,
+                0.0,
+                0.0,
+                gp,
+                false,
+                Collections.EMPTY_LIST,
+                false,
+                Collections.EMPTY_LIST,
+                "",
+                "",
+                "",
+                new HashMap());
+    }
+
+    @Test
+    public void copyConstructor() {
+        s1 = new Square();
```

Comment:
I know I haven't been mentioning this in the other reviews, but the naming for your variables shouldn't be so abstract.  You should name your variables "square1" instead of s1.  Take a moment and fix it for this class for all the variables